### PR TITLE
[PW-7487] Multiple giftcards 

### PR DESCRIPTION
**Pull request by amihajlovski**
This PR contains the implementation of supporting multiple gift cards while doing a payment.

Tested on the following scenarios:

1. Payment with two giftcards and 3ds2
2. Payment with two giftcards without 3ds
3. Payment with two giftcards and additional payment methods with multiple tabs open
4. Payment with two giftcards where cart amount is less than giftcards amount 
5. Changing the cart items, giftcard gets cancelled
6. Payment with a single giftcard, where full amount gets cover by the gift card
7. Payment with giftcard and additional payment method, while cancelling and re-adding the giftcard
8. Adding an item, applying giftcard, and then removing that item and adding other items, results on cancelation of the giftcard
9. Tested with 3 different giftcard brands and additional payment method including VVV with big amounts

### CODE REVIEW

**Review by zaiddreakh — DISMISSED** [review]

**Review by shanikantsingh — APPROVED** [review]

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/checkoutConfiguration.js [hunk 1]
```diff
@@ -180,9 +190,10 @@ function getGiftCardConfig() {
         },
       });
     },
-    onSubmit(state, component) {
+    onSubmit(state) {
       store.selectedMethod = state.data.paymentMethod.type;
-      store.brand = component?.displayName;
+      store.brand = brandArray?.filter((item, i, ar) => ar.indexOf(item) === i)
```
Comment (wboereboom) [inline src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/checkoutConfiguration.js:195]: Is there a way to make this lien a little easier to read?

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/renderGiftcardComponent.js [hunk 2]
```diff
@@ -44,18 +21,54 @@ function getGiftCardElements() {
...
+    const newListItem = document.createElement('li');
+    newListItem.setAttribute('data-brand', giftCard.brand);
+    newListItem.setAttribute('data-name', giftCard.name);
+    newListItem.setAttribute('data-type', giftCard.type);
+
+    const span = document.createElement('span');
+    span.textContent = giftCard.name;
+    const img = document.createElement('img');
+    img.src = `${imagePath}${giftCard.brand}.png`;
+    img.width = 40;
```
Comment (wboereboom) [inline src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/renderGiftcardComponent.js:47]: Will this cause issues on mobile?

File: src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/cancelPartialPaymentOrder.js [hunk 3]
```diff
@@ -22,20 +24,31 @@ function cancelPartialPaymentOrder(req, res, next) {
...
 
     if (response.resultCode === constants.RESULTCODES.RECEIVED) {
-      const currentBasket = BasketMgr.getCurrentBasket();
       Transaction.wrap(() => {
         collections.forEach(currentBasket.getPaymentInstruments(), (item) => {
           if (item.custom.adyenPartialPaymentsOrder) {
             currentBasket.removePaymentInstrument(item);
           }
         });
+        currentBasket.custom.adyenGiftCards = null;
```
Comment (wboereboom) [inline src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/cancelPartialPaymentOrder.js:33]: Should this stuff be moved to the clear function of Adyen Helper?

**Review by shanikantsingh — APPROVED** [review]

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/renderGiftcardComponent.js (hunk 2 above)
Comment (amihajlovski) [reply src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/renderGiftcardComponent.js:47]: This was a request from Cami, thats how the checkout team is doing for PMs as well.

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/checkoutConfiguration.js (hunk 1 above)
Comment (amihajlovski) [reply src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/checkoutConfiguration.js]: this was reverted to the previous implementation.

**Review by zaiddreakh — APPROVED** [review]

**Review by wboereboom — APPROVED** [review]

**Pull request merged**